### PR TITLE
Setup `@comet/brevo-mail-rendering` package

### PR DESCRIPTION
**Pull request by juliawegmayr**
## Description

Add site package

## Changeset

[ ] I have verified if my change requires a changeset

## Related tasks and documents

[COM-1033](https://vivid-planet.atlassian.net/browse/COM-1033)

## Further information

ImageBlock will be fully added in an upcoming PR. 

[COM-1033]: https://vivid-planet.atlassian.net/browse/COM-1033?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: demo/api/schema.gql [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
 # ------------------------------------------------------
 
-type User {
+type UserPermissionsUser {
```
[the remark this answers is not in the document]
Comment (juliawegmayr) [reply demo/api/schema.gql]: Fixed here: b35ebb0d460255364269bc0c150a33ef4c95929b

File: package.json [hunk 2]
```diff
@@ -67,5 +70,6 @@
                 }
             }
         }
-    }
+    },
+    "packageManager": "pnpm@9.14.2+sha512.6e2baf77d06b9362294152c851c4f278ede37ab1eba3a55fda317a4a17b209f4dbb973fb250a77abc463a341fcb1f17f17cfa24091c4eb319cda0d9b84278387"
```
Comment (johnnyomair) [inline package.json:74]: Why is this added?

File: packages/site/package.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        "lru-cache": "^11.0.2",
+        "next": "^14.2.18",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "redraft": "^0.10.2",
+        "rimraf": "^3.0.0",
+        "styled-components": "^6.1.13",
+        "swiper": "^11.1.15",
+        "ts-node": "^10.9.2"
+    },
```
Comment (johnnyomair) [inline packages/site/package.json:46]: I doubt you need any of this dependencies

File: packages/site/package.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    "devDependencies": {
```
Comment (johnnyomair) [inline packages/site/package.json:47]: The dev dependencies could probably be reduced as well.

**Review by johnnyomair — COMMENTED** [review]
What did you use as basis for the package? Demo Site, or `@comet/cms-site`?

File: packages/site/src/blocks/ImageBlock.tsx [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const ImageBlock = () => {
+    return null;
```
Comment (raphaelblum) [inline packages/site/src/blocks/ImageBlock.tsx:2]: Could we add generate-block-types in this Package as well ? (I'm not sure if this is easily achievable, but if so, it would be great.

https://github.com/vivid-planet/comet/blob/main/packages/site/cms-site/package.json#L19

File: package.json (hunk 2 above)
Comment (juliawegmayr) [reply package.json]: I don't know why this happened, but this line was added in the beginning, when I ran the install 🤔 
Anyway, I removed the line and it wasn't generated anymore: 47ad173c8ebb0383a8df44eb73c949ef80e02188

File: packages/site/package.json (hunk 3 above)
Comment (juliawegmayr) [reply packages/site/package.json]: Removed dependencies bdea97e28347a60ebeddd8c3f812fdec96a34128
and fixed error, because rimraf was missing: fcba8420c501b981353a0e2497c841074a9c4be6

File: packages/site/package.json (hunk 4 above)
Comment (juliawegmayr) [reply packages/site/package.json:29]: Also removed here: https://github.com/vivid-planet/comet-brevo-module/commit/bdea97e28347a60ebeddd8c3f812fdec96a34128

File: packages/site/package.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{
+    "name": "@comet/brevo-site",
+    "version": "1.0.0",
```
Comment (johnnyomair) [inline packages/site/package.json:3]: This version should be the same as the other packages.

File: packages/site/package.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        "dev": "tsc --watch --preserveWatchOutput -p tsconfig.build.json",
+        "generate-schema": "ts-node generate-schema.ts",
+        "generate-schema:watch": "chokidar \"src/\" -c \"$npm_execpath generate-schema\"",
+        "lint": "run-p lint:eslint lint:tsc",
+        "lint:eslint": "eslint --max-warnings 0 src/ package.json",
+        "lint:prettier": "npx prettier --check './**/*.{js,json,md,yml,yaml}'",
+        "lint:tsc": "tsc --noEmit"
+    },
+    "dependencies": {
+        "rimraf": "^3.0.0"
```
Comment (johnnyomair) [inline packages/site/package.json:27]: This can be a dev dependency. And please use the latest version.

File: packages/site/package.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    },
+    "devDependencies": {
+        "@cspell/eslint-plugin": "^8.16.0",
+        "eslint": "^8.0.0"
+    },
+    "peerDependencies": {
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "react-intl": "^5.0.0 || ^6.0.0",
+        "styled-components": "^6.0.0"
```
Comment (johnnyomair) [inline packages/site/package.json:37]: Peer dependenies need to be installed as dev dependency as well. Also, I'd remove react-intl and styled-components for now and only add the once they're needed.

File: packages/site/.eslintrc.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
+{
+    "extends": ["@comet/eslint-config/nextjs", "plugin:@cspell/recommended"],
```
Comment (johnnyomair) [inline packages/site/.eslintrc.json:2]: `@comet/eslint-config` is missing in the dev dependencies.

File: packages/site/package.json [hunk 10]
```diff
@@ -0,0 +1,43 @@
+{
+    "name": "@comet/brevo-site",
+    "version": "1.0.0",
+    "repository": {
+        "type": "git",
+        "url": "https://github.com/vivid-planet/comet",
```
Comment (johnnyomair) [inline packages/site/package.json:6]: This repository URL isn't correct.

File: packages/site/package.json [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    "files": [
+        "lib/*"
+    ],
+    "scripts": {
+        "build": "$npm_execpath run clean && tsc -p tsconfig.build.json",
+        "clean": "rimraf lib",
+        "dev": "tsc --watch --preserveWatchOutput -p tsconfig.build.json",
+        "generate-schema": "ts-node generate-schema.ts",
+        "generate-schema:watch": "chokidar \"src/\" -c \"$npm_execpath generate-schema\"",
+        "lint": "run-p lint:eslint lint:tsc",
```
Comment (johnnyomair) [inline packages/site/package.json:21]: lint:prettier should be included here as well.

File: packages/site/package.json [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    "types": "lib/index.d.ts",
+    "files": [
+        "lib/*"
+    ],
+    "scripts": {
+        "build": "$npm_execpath run clean && tsc -p tsconfig.build.json",
+        "clean": "rimraf lib",
+        "dev": "tsc --watch --preserveWatchOutput -p tsconfig.build.json",
+        "generate-schema": "ts-node generate-schema.ts",
+        "generate-schema:watch": "chokidar \"src/\" -c \"$npm_execpath generate-schema\"",
```
Comment (johnnyomair) [inline packages/site/package.json:20]: These don't work, we can remove them for now.

File: packages/site/src/blocks/ImageBlock.tsx (hunk 5 above)
Comment (johnnyomair) [reply packages/site/src/blocks/ImageBlock.tsx:2]: This will be done in a follow-up PR

**Comment by raphaelblum** [comment]
I don't think the name site fits. The code has nothing to do with a site/website. It is code to render e-mails.
It can be used in a standard comet site, but that doesn't have to be

Suggestion:
email-frontend?

File: packages/mail-rendering/.eslintrc.json [hunk 13]
```diff
@@ -0,0 +1,20 @@
+{
+    "extends": "@comet/eslint-config/nextjs",
```
Comment (johnnyomair) [inline packages/mail-rendering/.eslintrc.json:2]: We probably shouldn't use the nextjs config here, but react.

**Comment by johnnyomair** [comment]
> Or alternatively merge the setup into a feature branch and the feature branch has the changeset.

Maybe a feature branch would be a good idea anyway, since the package currently has no real integration anyway.

**Comment by juliawegmayr** [comment]
I added a feature branch for the package.

**Review by raphaelblum — DISMISSED** [review]

File: package.json [hunk 14]
```diff
@@ -65,6 +68,16 @@
...
                 }
+            },
+            "@types/hoist-non-react-statics": {
+                "peerDependencies": {
+                    "@types/react": "*"
+                }
+            },
+            "@types/react-transition-group": {
+                "peerDependencies": {
+                    "@types/react": "^17.0.0"
```
Comment (raphaelblum) [inline package.json:79]: why is it set types/react 17 and not react 18?

Comment (johnnyomair) [reply package.json:79]: This is for a package used in Admin, which is using React 17.

File: packages/mail-rendering/.eslintrc.json [hunk 15]
```diff
@@ -0,0 +1,20 @@
+{
+    "extends": "@comet/eslint-config/react",
+    "ignorePatterns": ["src/*.generated.ts", "lib/**"],
+    "rules": {
+        "@next/next/no-html-link-for-pages": "off", // disabled because lib has no pages dir
```
Comment (johnnyomair) [inline packages/mail-rendering/.eslintrc.json:5]:
```suggestion
```

This rule doesn't exist in the React config.

**Review by johnnyomair — CHANGES_REQUESTED** [review]

File: packages/mail-rendering/.eslintrc.json (hunk 15 above)
Comment (juliawegmayr) [reply packages/mail-rendering/.eslintrc.json]: a3a89e7d91d95e237e5bc14b361398d5bd2499ba

**Review by johnnyomair — APPROVED** [review]

**Review by raphaelblum — APPROVED** [review]
